File: RB101_Programming_Foundations/Small Probems/Easy6/right_triangles.rb
# input : a positive integer
# output : display a right triangle

# find the length of empty spaces to print, which is n - row number


# def triangle(rows)
#   row_number = 1

#   loop do
#     spaces = rows - row_number
#     puts "#{' ' * spaces}#{'*' * row_number}"
#     row_number += 1
#     break if row_number > rows
#   end
# end

# def triangle(rows)

#   for row_number in 1..rows
#     spaces = rows - row_number
#     puts "#{' ' * spaces}#{'*' * row_number}"
#   end
# end


def triangle(rows)
  rows.times do |row_number|
    spaces = rows - row_number
    puts "#{' ' * spaces}#{'*' * row_number}"
end
end
triangle(9)

File: RB101_Programming_Foundations/small_problems/Easy6/Python/right_triangles.py
# Write a method that takes a positive integer, n, as an argument, and displays a right triangle whose sides each have n stars. The hypotenuse of the triangle (the diagonal side in the images below) should have one end at the lower-left of the triangle, and the other end at the upper-right.

'''
Input: a number

Output: print the right triangle as shown

Rules


Data structure / Algorithms
    level starts at 1
    the number of spaces is the n - level

    iterate from 1 to (n + 1)
        print n - level number of spaces, then print level number of stars


'''

def triangle(n):
    for level in range(1, n + 1):
        print(f"{' ' * (n - level)}{'*' * level}")

triangle(5)
triangle(9)
